### PR TITLE
Eahw1624/annual target hours

### PR DESCRIPTION
**Pull request by IanFisherHO**

### CODE REVIEW

File: src/App.jsx [hunk 1]
```diff
@@ -1,33 +1,52 @@
...
+        <div className="App">
+          <Header />
+          <div className="govuk-width-container">
+            <main className="govuk-main-wrapper" id="main-content" role="main">
+              <ErrorBoundary>
+                <Routes>
+                  <Route index="true" element={<Home />} />
+                  <Route path="/timecard/:date" element={<Timecard />} />
+                  <Route path="/accruals/:date" element={<Accruals />} />
+                  <Route path="/forms" element={<FormsExample />} />
```
Comment (joannafawl) [inline src/App.jsx:34]: Could we remove this and the FormsExample component while we are amending the routing?

File: src/components/annual-target-hours/accruals/AnnualTargetHours.jsx [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+
+const formatToStringTime = (decimalHours) => {
+  if (decimalHours > 0) {
+    const hours = Math.trunc(decimalHours / 60);
+    let minutes = decimalHours % 60;
+    minutes = minutes.toString().padStart(2, '0');
+    return `${hours}:${minutes}`;
+  }
+  return '00:00';
+};
```
Comment (joannafawl) [inline src/components/annual-target-hours/accruals/AnnualTargetHours.jsx:78]: Should we move this into utils? Happy to leave it here if you think it should stay though

File: src/locales/en/common.json [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+            "name": "beta",
+            "feedback": "This is a new service - your <feedback_anchor>feedback</feedback_anchor> will help us to improve it."
+        }
+    },
+    "navigation": {
+        "home": "Home"
+    },
+    "accruals": {
+        "balance": "My accruals balance as of",
+        "dateFormat": "D MMMM YYYY",
```
Comment (joannafawl) [inline src/locales/en/common.json:13]: Does this need to be kept in here?

File: src/pages/accruals/Accruals.jsx [hunk 4]
```diff
@@ -0,0 +1,168 @@
+/* eslint-disable prettier/prettier */
```
Comment (joannafawl) [inline src/pages/accruals/Accruals.jsx:1]: How come prettier has been disabled?

File: src/pages/accruals/Accruals.jsx [hunk 5]
```diff
@@ -0,0 +1,168 @@
...
+    else {
+      await getAccrualsData(
+        accrualsDate,
+        setServiceError
+      )
+    }
+  }, [accrualsDate]);
+
+  const isOutsideAgreementDates = () => {
+
```
Comment (joannafawl) [inline src/pages/accruals/Accruals.jsx:46]: Minor but could we remove this line?

File: src/pages/accruals/Accruals.jsx [hunk 6]
```diff
@@ -0,0 +1,168 @@
...
+ 
+  return (
+    <>
+      <h1 className="govuk-caption-m">{t('accruals.balance')}</h1>
+      <h2 className="govuk-heading-m">
+      {dayjs(accrualsDate).format(t('accruals.dateFormat'))}
+      </h2>
+      <div className="govuk-button-group button-group-row">
+        <Link
+          onClick={() => {}}
```
Comment (joannafawl) [inline src/pages/accruals/Accruals.jsx:149]: Is this needed if it's empty?

File: src/pages/accruals/Accruals.spec.jsx [hunk 7]
```diff
@@ -0,0 +1,157 @@
...
+        defaultApplicationContext,
+        '/2022-07-01',
+        '/:date'
+      );
+
+      const previousDayLink = getByRole('link', {
+        name: t('accruals.previousDay'),
+      });
+      expect(previousDayLink.pathname).toBe('/accruals/2022-06-30');
+      fireEvent.click(previousDayLink);
```
Comment (joannafawl) [inline src/pages/accruals/Accruals.spec.jsx:60]: Why do we click the link after the expect statement?

File: src/components/annual-target-hours/accruals/AnnualTargetHours.jsx [hunk 8]
```diff
@@ -0,0 +1,104 @@
...
+  const target = total - (accrualsData?.cumulativeTarget || 0);
+
+  return (
+    <div className="accruals-container">
+      <form className="grey-border">
+        <h1 className="govuk-heading-m newline">
+          {targetData ? (
+            <span
+              dangerouslySetInnerHTML={{
+                __html: t('annualTargetHours.titleRemaining', {
```
Comment (JWeston-HO) [inline src/components/annual-target-hours/accruals/AnnualTargetHours.jsx:20]: NIT: Could we rename titleRemaining to remainingHoursTitle?

File: src/components/annual-target-hours/accruals/AnnualTargetHours.jsx (hunk 2 above)
Comment (JWeston-HO) [reply src/components/annual-target-hours/accruals/AnnualTargetHours.jsx:78]: Second this.

File: src/App.jsx (hunk 1 above)
Comment (IanFisherHO) [reply src/App.jsx]: Removed FormsExample from routing and code

File: src/components/annual-target-hours/accruals/AnnualTargetHours.jsx [hunk 9]
```diff
@@ -0,0 +1,104 @@
+import PropTypes from 'prop-types';
+import { useTranslation } from 'react-i18next';
+import '../../../i18n';
+
+const AnnualTargetHours = ({ targetData, accrualsData }) => {
+  const { t } = useTranslation('common');
+  const total = targetData?.targetTotal || 0;
+  const worked = accrualsData?.cumulativeTotal || 0;
+  const remainingMins = Math.floor(total - worked);
+  const remainingWhole = Math.floor(remainingMins / 60);
```
[the remark this answers is not in the document]
Comment (IanFisherHO) [reply src/components/annual-target-hours/accruals/AnnualTargetHours.jsx]: Renamed variable

File: src/components/annual-target-hours/accruals/AnnualTargetHours.jsx (hunk 2 above)
Comment (IanFisherHO) [reply src/components/annual-target-hours/accruals/AnnualTargetHours.jsx]: Moved to utils and extra tests added. Also renamed input.

File: src/pages/accruals/Accruals.jsx (hunk 4 above)
Comment (IanFisherHO) [reply src/pages/accruals/Accruals.jsx]: Removed

File: src/locales/en/common.json [hunk 10]
```diff
@@ -0,0 +1,26 @@
...
+        "previousDay": "Previous day",
+        "nextDay": "Next day"
```
[the remark this answers is not in the document]
Comment (IanFisherHO) [reply src/locales/en/common.json]: Moved to navigation section

File: src/pages/accruals/Accruals.jsx (hunk 5 above)
Comment (IanFisherHO) [reply src/pages/accruals/Accruals.jsx]: Fixed by prettier

File: src/pages/accruals/Accruals.jsx [hunk 11]
```diff
@@ -0,0 +1,168 @@
...
+
+    return ((agreementStartDate === null || agreementEndDate === null) ||
+    (accrualsDate < agreementStartDate || accrualsDate > agreementEndDate));
+  };
+
+  const filterAccrualsData = (fetchedAccruals) => {
+    fetchedAccruals.forEach(accrual => {
+
+      if (accrual.accrualTypeId === ANNUAL_TARGET_HOURS_ID)
+      setAnnualTargetHoursData(accrual);
```
[the remark this answers is not in the document]
Comment (IanFisherHO) [reply src/pages/accruals/Accruals.jsx]: Brackets added

File: src/pages/accruals/Accruals.jsx (hunk 6 above)
Comment (IanFisherHO) [reply src/pages/accruals/Accruals.jsx]: Removed on click

File: src/pages/accruals/Accruals.spec.jsx (hunk 7 above)
Comment (IanFisherHO) [reply src/pages/accruals/Accruals.spec.jsx]: Removed click

File: src/pages/accruals/Accruals.spec.jsx [hunk 12]
```diff
@@ -0,0 +1,157 @@
...
+        '/:date'
+      );
+
+      const nextDayLink = getByRole('link', { name: t('accruals.nextDay') });
+      expect(nextDayLink.pathname).toBe('/accruals/2022-07-02');
+      fireEvent.click(nextDayLink);
+    });
+  });
+
+  describe('getAcrrualsData', () => {
```
[the remark this answers is not in the document]
Comment (IanFisherHO) [reply src/pages/accruals/Accruals.spec.jsx]: Fixed typo

File: src/components/annual-target-hours/accruals/AnnualTargetHours.jsx (hunk 8 above)
Comment (IanFisherHO) [reply src/components/annual-target-hours/accruals/AnnualTargetHours.jsx]: Renamed to more suitable variable name

**Review by joannafawl — APPROVED** [review]

**Review by joannafawl — APPROVED** [review]

**Review by JWeston-HO — APPROVED** [review]

**Pull request merged**